### PR TITLE
[FIX] purge uninstalled uninstallable modules instead of deleting

### PR DESCRIPTION
**Pull request by hbrunn**
forward port of https://github.com/OCA/server-tools/pull/489

### CODE REVIEW

**Comment by dreispt** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ addon | version | summary
 [auth_signup_verify_email](auth_signup_verify_email/) | 9.0.1.0.0 | Force uninvited users to use a good email for signup
 [auth_supplier](auth_supplier/) | 9.0.2.0.0 | Auth Supplier
 [base_custom_info](base_custom_info/) | 9.0.1.0.0 | Add custom field in models
-[base_multi_image](base_multi_image/) | 9.0.1.0.0 | Allow multiple images for database objects
+[base_multi_image](base_multi_image/) | 9.0.1.1.0 | Allow multiple images for database objects
```
Comment (moylop260) [inline README.md:25]: FYI the section `[//]: # (addons)` is auto-generated you don't need add this change.

**Comment by moylop260** [comment]
👍

File: README.md (hunk 1 above)
Comment (pedrobaeza) [inline README.md:25]: In fact, this generates conflicts continuously as now, so please remove it and squash the commits

**Comment by gurneyalex** [comment]
@hbrunn rebase needed.

**Comment by hbrunn** [comment]
did someone force-push over the 9.0 branch? My local copy conflicted with it when fetching...

**Comment by gurneyalex** [comment]
@hbrunn I think @pedrobaeza may have done so, yes (https://twitter.com/PedroMBaeza/status/761595556934410240)

**Comment by pedrobaeza** [comment]
Yes, I did. See https://github.com/OCA/server-tools/issues/504

**Comment by hbrunn** [comment]
ah, thanks. Missed that while enjoying European High Culture: http://www.wacken.com ;-)

**Comment by hbrunn** [comment]
now investigating the CI issues, I have the feeling isolating this addon in the tests rather fixes a symptom than the cause

**Comment by gurneyalex** [comment]
@hbrunn did you see #507 and #508?

**Comment by hbrunn** [comment]
@gurneyalex yes, thanks. But those work by isolating the module, and I'd like to find out what actually goes wrong in the first place.

**Comment by StefanRijnhart** [comment]
:+1: Any luck finding out the problem of testing this module in combination with other modules?

**Comment by hbrunn** [comment]
@StefanRijnhart yes, the test introduced in the offending commit caused the registry to be reloaded, with quite undefined behavior afterwards because at this point, there's no change signalling and no reloading. So I had to do some trickery to reload the registry, but not really: https://github.com/OCA/server-tools/pull/512/files

**Pull request merged**